ui/helpers/utils/icon-factory.js

```diff
@@ -18,7 +19,7 @@ function IconFactory(jazzicon) {
 IconFactory.prototype.iconForAddress = function (address, diameter) {
   let addr = address;
 
-  if (isHex(address)) {
+  if (isValidHexAddress(address, false)) {
```

Comment:
This seems dangerous. Previously `checksumAddress` would add the prefix, and checksum the address regardless of whether it's mixed-case or not. Now it requires the address is already hex-prefixed and either checksummed already or all one case. This means for any other case, we're skipping this normalization step, and we'll be generating a different icon than before.

Reply:
under the hood, the utility method 'checksumAddress' calls ethereumjs-util's toChecksumAddress which in 5.1 adds the '0x' prefix, and in the new version, it throws an error. It seems like we should probably add a new utility method that mimics this behavior because as is even when allowing this to validate true for a non hex prefixed string, the string is not modified meaning it would immediately throw on the next line.

I'm going to add this utility method now, and replace usages of the old helper method.

Reply:
actually I should do that in a new PR instead to prevent convoluting this with new logic